src/runtime_stack_check.js

```diff
@@ -13,22 +13,23 @@ function writeStackCookie() {
   // The stack grows downwards
   HEAPU32[(STACK_MAX >> 2)+1] = 0x2135467;
   HEAPU32[(STACK_MAX >> 2)+2] = 0x89BACDFE;
-#if !USE_ASAN
+#if !USE_ASAN && !SAFE_HEAP // ASan and SAFE_HEAP check address 0 themselves
   // Also test the global address 0 for integrity.
-  // We don't do this with ASan because ASan does its own checks for this.
   HEAP32[0] = 0x63736d65; /* 'emsc' */
 #endif
 }
 
 function checkStackCookie() {
+#if !MINIMAL_RUNTIME
+  if (ABORT) return;
```

Comment:
If you are going to add this here you can remove the this check from `doRun` where it calls `checkStackCookie` at the end

Reply:
done